ingest.py

```diff
@@ -1,316 +0,0 @@
-
-    def _insert_gamelogs(self, gamelogs):
-        """
-        Adds gamelogs to the database.
-
-        :param gamelogs: Gamelogs, each being a dict of stat type to stat value,
-            to add to the database.
-        """
-        db.headtoheads.insert(
-            self._update_gamelog_keys(gamelog) for gamelog in gamelogs)
-
-    def _update_gamelog_keys(self, gamelog):
-        """
-        Removes Player key and switches MainPlayerCode and OppPlayerCode
-        keys if the MainPlayerCode is not player_code.
-
-        :param gamelog: A dict of stats.
-        :returns: Gamelog with proper keys placement.
-        """
-        # @TODO This is so sad.
-        gamelog.pop('Player', None)
-        if self.player_one_code != gamelog['MainPlayerCode']:
-            replacer = partial(multiple_replace,
-                               adict={'Main': 'Opp', 'Opp': 'Main'})
-            return walk_keys(replacer, gamelog)
-        return gamelog
-
-
-class PlayerIngester(object):
-    _br_url = 'http://www.basketball-reference.com'
-    _date_regex = re.compile('\d{4}-\d{2}')
-
-    def __init__(self, letter):
-        self._letter = letter
-
-    def find(self):
-        """
-        Finds the home urls for all players whose last names start with a
-        particular letter.
-        """
-        # Current players are noted with bold text.
-        url = "{self._br_url}/players/{self._letter}".format(self=self)
-        letter_page = requests.get(url).text
-        current_names = BeautifulSoup(letter_page).find_all('strong')
-        if current_names:
-            db.players.insert(
-                self._create_player_dict(name) for name in current_names)
-
-    def _get_gamelog_urls(self, player_url):
-        """
-        Returns list of gamelog urls with every year for one player.
-
-        :param player_url: Url for a player's basketball-reference profile.
-        :returns: List of basketball-reference urls for each year of a player's
-            career.
-        """
-        player_page = requests.get(player_url).text
-        tables = (BeautifulSoup(player_page)
-                  .find('table', {'id': 'totals'})
-                  .find_all('tr', {'class': 'full_table'}))
-        return [
-            self._br_url + link.get('href')
-            for table in tables
-            for link in table.find('td').find_all('a')
-            if self._date_regex.match(str(link.get_text()))
-        ]
-
-    def _create_player_dict(self, name):
-        """
-        Create a dictionary for a player to enter into the database.
-
-        :param name: Name of the player.
-        :returns: Dictionary of player data to add to database.
-        """
-        name_data = next(name.children)
-        player_url = self._br_url + name_data.attrs['href']
-        return {
-            'Player': str(name_data.contents[0]),
-            'GamelogURLs': self._get_gamelog_urls(player_url),
-            'URL': player_url
-        }
```

Comment:
Not that I dislike the functional programming style, but it makes it slightly harder to parse the code since it uses a lot of external libraries to process the data. Tradeoffs to think about.